src/d3a/models/area/__init__.py

```diff
@@ -146,45 +138,35 @@ def area_reconfigure_event(self, **kwargs):
-            else self.baseline_peak_energy_import_kWh
+            else self.throughput.baseline_peak_energy_import_kWh
 
         baseline_peak_energy_export_kWh = kwargs["baseline_peak_energy_export_kWh"] \
             if key_in_dict_and_not_none(kwargs, 'baseline_peak_energy_export_kWh') \
-            else self.baseline_peak_energy_export_kWh
+            else self.throughput.baseline_peak_energy_export_kWh
 
-        if key_in_dict_and_not_none(kwargs, 'import_capacity_kVA'):
-            import_capacity_kVA = kwargs["import_capacity_kVA"]
-            import_capacity_kWh = convert_area_throughput_kVA_to_kWh(import_capacity_kVA,
-                                                                     self.config.slot_length)
-        else:
-            import_capacity_kVA = None
-            import_capacity_kWh = self.import_capacity_kWh
+        import_capacity_kVA = kwargs["import_capacity_kVA"] \
+            if key_in_dict_and_not_none(kwargs, 'import_capacity_kVA') \
+            else self.throughput.import_capacity_kVA
 
-        if key_in_dict_and_not_none(kwargs, 'export_capacity_kVA'):
-            export_capacity_kVA = kwargs["export_capacity_kVA"]
-            export_capacity_kWh = convert_area_throughput_kVA_to_kWh(export_capacity_kVA,
-                                                                     self.config.slot_length)
-        else:
-            export_capacity_kVA = None
-            export_capacity_kWh = self.export_capacity_kWh
+        export_capacity_kVA = kwargs["export_capacity_kVA"] \
+            if key_in_dict_and_not_none(kwargs, 'export_capacity_kVA') \
+            else self.throughput.import_capacity_kVA
 
         try:
             validate_area(grid_fee_constant=grid_fee_constant,
-                          grid_fee_percentage=grid_fee_percentage,
-                          baseline_peak_energy_import_kWh=baseline_peak_energy_import_kWh,
-                          baseline_peak_energy_export_kWh=baseline_peak_energy_export_kWh,
-                          import_capacity_kVA=import_capacity_kVA,
-                          export_capacity_kVA=export_capacity_kVA)
+                          grid_fee_percentage=grid_fee_percentage)
 
         except Exception as e:
             log.error(str(e))
             return
 
         self._set_grid_fees(grid_fee_constant, grid_fee_percentage)
-        self.baseline_peak_energy_import_kWh = baseline_peak_energy_import_kWh
-        self.baseline_peak_energy_export_kWh = baseline_peak_energy_export_kWh
-        self.export_capacity_kWh = export_capacity_kWh
-        self.import_capacity_kWh = import_capacity_kWh
+        self.throughput = ThroughputParameters(
+            baseline_peak_energy_import_kWh=baseline_peak_energy_import_kWh,
+            baseline_peak_energy_export_kWh=baseline_peak_energy_export_kWh,
+            import_capacity_kVA=import_capacity_kVA,
+            export_capacity_kVA=export_capacity_kVA
+        )
```

Comment:
ThroughputParameters class also validates that the parameters are correct. This might introduce a slight bug here, because if this validation in thje ThroughputParameters fails, the _set_grid_fees would have already changed the grid fees, thus updating partially the state of the area class (grid fees would be updated, throughput parameters would not be updated). I would recommend to  circumvent this by constructing the ThroughputParameters object in the try/catch block above, and only assign it to self.throughput here.

Reply:
Indeed! Thanks for finding this! Will do.